pkg/controller/registry/grpc/source.go

```diff
@@ -200,11 +202,12 @@ func (s *SourceStore) AsClients(namespaces ...string) map[resolver.CatalogKey]cl
 		}
 		for _, namespace := range namespaces {
 			if key.Namespace == namespace {
-				refs[key] = client.NewClientFromConn(source.Conn)
+				refsInterface[key] = client.NewClientFromConn(source.Conn)
+				refsClient[key] = registry.NewRegistryClient(client.NewClientFromConn(source.Conn))
```

Comment:
If we [use composition](https://github.com/operator-framework/operator-lifecycle-manager/pull/1521/files#r428138508), we can avoid passing around two clients.